### PR TITLE
Cut over to the new spec authoring platform

### PR DESCRIPTION
**Pull request by turt2live**
Should check off enough of https://github.com/matrix-org/matrix-doc/issues/3055 to unfreeze. Incorporates much of https://github.com/matrix-org/matrix-doc/pull/3025, though we may still want to look at https://github.com/matrix-org/matrix-doc/pull/3025 for supplemental documentation.

In collaboration with https://github.com/matrix-org/matrix.org/pull/970

----

**Reviewer**: The commits are largely isolated and have a bit of a story progression to them. This isn't a perfect cleanup, but it should be enough to continue spec writing.

----

The CI is not transitioned fully here, but the now-wasted bytes from the bundle are removed.

### CODE REVIEW

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 steps:
```
Comment (anoadragon453) [inline .buildkite/pipeline.yaml:1]: There's a couple places where we should update links to documents in the `drafts/` folder.

* https://github.com/matrix-org/matrix.org/blob/e5d5c515fd103edc14b650e62d70c11735b42090/gatsby/src/pages/index.js#L277
* https://github.com/matrix-org/matrix-doc/issues/1148#issue-302742197

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -100,8 +103,7 @@ the ``newsfragments`` directory. The ``type`` can be one of the following:
 
 All news fragments must have a brief summary explaining the change in the
 contents of the file. The summary must end in a full stop to be in line with
-the style guide and and formatting must be done using `Restructured Text
-<http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_.
+the style guide and and formatting must be done using Markdown.
```
Comment (anoadragon453) [inline CONTRIBUTING.rst:106]:
```suggestion
the style guide and formatting must be done using Markdown.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  For example, Sass files would go here.
+
+* `/content`: files that will become pages in the site go here. Typically these are Markdown files with some YAML front 
+  matter indicating, [among other things](https://gohugo.io/content-management/front-matter/), what layout should be 
+  applied to this page. The organization of files under `/content` determines the organization of pages in the built 
+  site.
+
+* `/data`: this can contain TOML, YAML, or JSON files. Files kept here are directly available to template code as 
+  [data objects](https://gohugo.io/templates/data-templates/), so templates don't need to load them from a file and 
+  parse them. This is also where our
```
Comment (anoadragon453) [inline README.md:24]: Where our what?!1?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      template code that's used in more than one template. An obvious example here is something like a sidebar, where 
+      several different page layouts might all include the sidebar. But also, partial templates can return values: this 
+      means they can be used like functions, that can be called by multiple templates to do some common processing.
+    * `/layouts/shortcodes`: these templates can be called directly from files in `/content`.
+
+* `/static`: static files which don't need preprocessing. JS or CSS files could live here.
+
+* `/themes`: you can use just Hugo or use it with a theme. Themes primarily provide additional templates, which are 
+  supplied in a `/themes/$theme_name/layouts` directory. You can use a theme but customise it by providing your own 
+  versions of any of the them layouts in the base `/layouts` directory. That is, if a theme provides
```
Comment (anoadragon453) [inline README.md:38]:
```suggestion
  versions of any of the theme layouts in the base `/layouts` directory. That is, if a theme provides 
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  defined here are available in templates. All these directories above are configurable via `config.toml` settings.
+
+Additionally, the following directories may be of interest:
+
+* `/attic`: Here contains historical sections of specification and legacy drafts for the specification.
+* `/changelogs`: Various bits of changelog for the specification areas.
+* `/event-schemas`: [JSON Schema](http://json-schema.org/) definitions for the spec.
+* `/data-definitions`: Bits of structured data consumable by Matrix implementations.
+* `/meta`: Documentation relating to the spec's processes that are otherwise untracked (release instructions, etc).
+* `/scripts`: Various scripts for generating the spec.
```
Comment (anoadragon453) [inline README.md:54]:
```suggestion
* `/scripts`: Various scripts for generating the spec and validating its contents.
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+## Authoring changes to the spec
+
+Please read [CONTRIBUTING.rst](./CONTRIBUTING.rst) before authoring a change to the spec. Note that spec authoring takes
+place after an MSC has been accepted, not as part of a proposal itself.
+
+1. Install the extended version (often the OS default) of Hugo: <https://gohugo.io/getting-started/installing>
+2. Run `git submodule update --init --recursive` for good measure.
+3. Run `npm i` to install the dependencies. Note that this will require NodeJS to be installed.
+4. Run `npm run get-proposals` to seed the proposals data. This is not required.
```
Comment (anoadragon453) [inline README.md:67]:
```suggestion
4. Run `npm run get-proposals` to seed proposal data. This is merely for populating the content of the "Spec Change Proposals" page and is not required.
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+5. Run `hugo serve` to run a local webserver which builds whenever a file change is detected. If watching doesn't appear
+   to be working for you, try `hugo serve --disableFastRender` instead.
+6. Edit the specification 🙂
+
+We use a highly customized [Docsy](https://www.docsy.dev/) theme for our generated site, which uses Bootstrap and Font 
+Awesome. If you're looking at making design-related changes to the spec site, please coordinate with us in 
+[#matrix-docs:matrix.org](https://matrix.to/#/#matrix-docs:matrix.org) before opening a PR.
```
Comment (anoadragon453) [inline README.md:74]: Not sure if we care, but for consistency we use `http://` matrix.to links above (we could also go the other way and use https:// :innocent:).

File: README.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+## Building the specification
+
+If for some reason you're not a CI/CD system and want to render the spec yourself, follow the above steps for authoring
```
Comment (anoadragon453) [inline README.md:78]:
```suggestion
If for some reason you're not a CI/CD system and want to render a static version of the spec for yourself, follow the above steps for authoring
```

File: README.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+For building the swagger definitions, create a python3 virtualenv and activate it. Then run `pip install -r ./scripts/requirements.txt` and finally `python ./scripts/dump-swagger.py` to generate it to `./scripts/swagger/api-docs.json`.
+To make use of the generated file, there are a number of options:
+
+* It can be uploaded from your filesystem to an online editor/viewer such as
+  http://editor.swagger.io/
+* You can run a local HTTP server by running `./scripts/swagger-http-server.py`, and then view the documentation via an
+  online viewer; for example, at <http://petstore.swagger.io/?url=http://localhost:8000/api-docs.json>
+* You can host the swagger UI yourself. See <https://github.com/swagger-api/swagger-ui#how-to-run> for advice on how to 
+  do so.
```
Comment (anoadragon453) [inline README.md:91]:
```suggestion
  do so
```

File: meta/documentation_style.rst [hunk 10]
```diff
@@ -8,52 +8,36 @@ in.
 Format
 ------
 
-Documentation is written either in github-flavored markdown or RST.
+Documentation is written either in github-flavored markdown.
```
Comment (anoadragon453) [inline meta/documentation_style.rst:11]:
```suggestion
Documentation is written in github-flavored markdown.
```

File: .buildkite/pipeline.yaml (hunk 1 above)
Comment (anoadragon453) [inline .buildkite/pipeline.yaml:1]: We should also update Gilesbot's config to remove the legacy spec lines: https://github.com/matrix-org/matrix-doc/blob/831bd91ce931b80c6fc35f6d5e74f9daa140138b/pyproject.toml#L3-L5

Guilesbot is what generates the little "Click here to view the spec" checks on PRs. Note that you won't see the results of this change until it is in `master`.

File: .circleci/config.yml [hunk 11]
```diff
@@ -54,12 +48,6 @@ buildspeculator: &buildspeculator
     cd scripts/speculator
     go build -v
```
Comment (anoadragon453) [inline .circleci/config.yml:49]: I wonder if we need this as part of CI anymore?

File: README.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+      several different page layouts might all include the sidebar. But also, partial templates can return values: this 
+      means they can be used like functions, that can be called by multiple templates to do some common processing.
+    * `/layouts/shortcodes`: these templates can be called directly from files in `/content`.
+
+* `/static`: static files which don't need preprocessing. JS or CSS files could live here.
+
+* `/themes`: you can use just Hugo or use it with a theme. Themes primarily provide additional templates, which are 
+  supplied in a `/themes/$theme_name/layouts` directory. You can use a theme but customise it by providing your own 
+  versions of any of the them layouts in the base `/layouts` directory. That is, if a theme provides 
+  `/themes/$theme_name/layouts/sidebar.html` and you provide `/layouts/sidebar.html`, then your version of this
```
Comment (anoadragon453) [inline README.md:39]:
```suggestion
  `/themes/$theme_name/layouts/sidebar.html` and you provide `/layouts/sidebar.html`, then your version of the 
```

File: README.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+Developers looking to use Matrix should join [#matrix-dev:matrix.org](http://matrix.to/#/#matrix-dev:matrix.org)
+on Matrix for help.
+
+Spec authors and proposal writers are welcome to join [#matrix-spec:matrix.org](http://matrix.to/#/#matrix-spec:matrix.org). We welcome contributions! See [CONTRIBUTING.rst](./CONTRIBUTING.rst) for details.
+
+## Structure
+
+The Matrix spec is compiled with [Hugo](https://gohugo.io/) (a static site generator) with the following structure:
+
+* `/assets`: assets that need postprocessing using [Hugo Pipes](https://gohugo.io/hugo-pipes/introduction/).
```
Comment (anoadragon453) [inline README.md:14]: Can we rename the current `assets-hugo` folder to `assets`? (Other bits might need updating.)

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
This looks great on the whole. Just a bunch of little fixes and suggestions below.

File: README.md (hunk 3 above)
Comment (turt2live) [reply README.md:24]: we may never know

File: .circleci/config.yml (hunk 11 above)
Comment (turt2live) [reply .circleci/config.yml:49]: probably not, but it doesn't seem to be tested elsewhere. Phase N+1 will take care of it

File: README.md (hunk 13 above)
Comment (turt2live) [reply README.md:14]: I'd advocate for doing that in another PR for consideration.

File: README.md (hunk 7 above)
Comment (turt2live) [reply README.md:74]: I'm going to change them to https. There's zero point in 2021 to use http.

File: README.md (hunk 9 above)
Comment (turt2live) [reply README.md:91]: This suggestion goes against our documentation style guide.

Comment (anoadragon453) [reply README.md:91]: Oh, right. Well ten perhaps add periods to the other bullet point lines for consistency?

Comment (turt2live) [reply README.md:91]: but then they break the links :D 

Our consistency says this is fine, but happy to look at rewording the text to not have a link at the end

File: scripts/speculator/main.go [hunk 14]
```diff
@@ -337,7 +337,7 @@ func (s *server) serveSpec(w http.ResponseWriter, req *http.Request) {
 		log.Printf("Serving pr %s (%s)\n", branchName, sha)
 	} else if strings.ToLower(branchName) == "head" ||
 		branchName == "master" ||
-		strings.HasPrefix(branchName, "drafts/") {
+		strings.HasPrefix(branchName, "attic/drafts/") {
```
Comment (anoadragon453) [inline scripts/speculator/main.go:340]: Good catch!

**Review by anoadragon453 — APPROVED** [review]

File: scripts/speculator/main.go (hunk 14 above)
Comment (turt2live) [reply scripts/speculator/main.go:340]: ctrl+f for the win

**Pull request merged**